packages/sample-app/cypress.config.ts

```diff
@@ -0,0 +1,24 @@
+import { defineConfig } from 'cypress';
+import webpackConfig from './webpack.config';
+
+export default defineConfig({
+  screenshotsFolder: '../../screenshots',
+  video: false,
+
+  component: {
+    devServer: {
+      framework: 'react',
+      bundler: 'webpack',
+      webpackConfig,
+    },
+    specPattern: 'src/components/**/*.cy.{js,jsx,ts,tsx}',
+    indexHtmlFile: 'src/cypress/component-index.html',
```

Comment:
NIT: Wondering if it makes sense to put the `cypress` and `e2e` directories in the `src` folder or if we should move them out to top level directories.

Reply:
We currently put all the source code in `src` directory per each package.

I don't see much value in creating sibling source directories scoped to specific tools or purposes - having all the sources under one directory seems simpler to me. (Test code is source code too, but oftentimes get treated as second class citizen in the codebase.)

AFAK there is no special treatment necessary for `src/cypress` directory - its code is used in the Cypress component testing webpack build, so the usual context applies (React code to be part of the resulting web application).

I've placed `component-index.html` into the `cypress` directory to co-locate the HTML template with associated setup code. This is similar to `src/app-index.html.ejs` and `src/app-styles.css` sitting next to `src/app.tsx`.

Note to self: `app.css` could be inlined to `app-styles.css` and the latter renamed to `app.css`.

As for `src/e2e` - this was meant to contain all E2E tests. If you think that e.g. `src/app.e2e.cy.ts` is better, I can change that.